src/apis/orderbookapi.py

```diff
@@ -0,0 +1,123 @@
+        """
+        Get solver competition data from a transaction hash.
+        The returned dict follows the schema outlined here:
+        https://api.cow.fi/docs/#/default/get_api_v1_solver_competition_by_tx_hash__tx_hash_
+        """
+        prod_endpoint_url = f"{PROD_BASE_URL}solver_competition/by_tx_hash/{tx_hash}"
+        barn_endpoint_url = f"{BARN_BASE_URL}solver_competition/by_tx_hash/{tx_hash}"
+        try:
+            json_competition_data = requests.get(
+                prod_endpoint_url,
+                headers=header,
+                timeout=REQUEST_TIMEOUT,
+            )
+            if json_competition_data.status_code == SUCCESS_CODE:
+                solver_competition_data = json.loads(json_competition_data.text)
+            elif json_competition_data.status_code == FAIL_CODE:
+                barn_competition_data = requests.get(
+                    barn_endpoint_url, headers=header, timeout=REQUEST_TIMEOUT
+                )
+                if barn_competition_data.status_code == SUCCESS_CODE:
+                    solver_competition_data = json.loads(barn_competition_data.text)
+                else:
+                    return None
+        except requests.exceptions.ConnectionError as err:
+            self.logger.warning(
+                f"Connection error while fetching competition data: {err}"
+            )
+            return None
+        return solver_competition_data
+
+    def get_quote(self, trade: Trade) -> Optional[Trade]:
+        """
+        Given a trade, compute buy_amount, sell_amount, and fee_amount of the trade
+        as proposed by our quoting infrastructure.
+        """
+
+        if trade.data.is_sell_order:
+            kind = "sell"
+            limit_amount_name = "sellAmountBeforeFee"
+            executed_amount = trade.execution.sell_amount
+        else:
+            kind = "buy"
+            limit_amount_name = "buyAmountAfterFee"
+            executed_amount = trade.execution.buy_amount
+
+        request_dict = {
+            "sellToken": trade.data.sell_token,
+            "buyToken": trade.data.buy_token,
+            "receiver": "0x0000000000000000000000000000000000000000",
+            "appData": "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "partiallyFillable": False,
+            "sellTokenBalance": "erc20",
+            "buyTokenBalance": "erc20",
+            "from": "0x0000000000000000000000000000000000000000",
+            "priceQuality": "optimal",
+            "signingScheme": "eip712",
+            "onchainOrder": False,
+            "kind": kind,
+            limit_amount_name: str(executed_amount),
+        }
+        prod_endpoint_url = f"{PROD_BASE_URL}quote"
+
+        try:
+            quote_response = requests.post(
+                prod_endpoint_url,
+                headers=header,
+                json=request_dict,
+                timeout=REQUEST_TIMEOUT,
+            )
+        except ValueError as err:
```

Comment:
Again, is it possible that we get some other exception here and the script "crashes"?

Reply:
Not sure anymore why I used a different error here. Will have a look at what types of errors are possible and catch 'em all. (In the sense of 'what types of errors are typically returned by a post request'. In python, anything can happen a run time.)

Reply:
I changed the error type.